src/QMCWaveFunctions/lcao/LCAOrbitalBuilder.cpp

```diff
@@ -698,7 +705,30 @@ bool LCAOrbitalBuilder::putPBCFromH5(LCAOrbitalSet& spo, xmlNodePtr coeff_ptr)
   spo.setIdentity(false);
   hdf_archive hin(myComm);
 
-  xmlNodePtr curtemp = coeff_ptr->parent->parent->parent;
+  xmlNodePtr curtemp = coeff_ptr;
+
+  std::string xmlTag("determinantset");
+  std::string MSDTag("sposet");
+  std::string SDTag("determinant");
+  std::string EndTag("qmcsystem");
+  std::string curname;
+
+  do
+  {
+    std::stringstream ss;
+    curtemp = curtemp->parent;
+    ss << curtemp->name;
+    ss >> curname;
+    if (curname == MSDTag)
+      MultiDet = true; ///Used to know if running an MSD calculation - needed for order of Orbitals.
+    if (curname == SDTag)
+      MultiDet = false;
+
+  } while ((xmlTag != curname) && (curname != EndTag));
+  if (curname == EndTag)
+    APP_ABORT(
```

Comment:
Macros should be encapsulated in braces {}.  No guarantee of being single line.

Reply:
How will the EndTag check prevent the infinite loop?  Aren't the tags like:

<qmcsystem>
  <wavefunction>
    ...
  </wavefunction>
</qmcsystem>

The </qmcsystem> tag is not ever available via cur->name.

Reply:
Please explicitly try running the code on an input that has all `<determinant/>` and `<sposet/>` elements removed (and at least has some others, say a typo: `<spset/>`.

I think this case will result in an infinite loop with the current code.

Reply:
APP_ABORT is protected properly by curly bracket.